support for 32 bit platform

I get the following npm-debug.log.

[npm-debug.txt](https://github.com/bloomberg/bucklescript/files/713827/npm-debug.txt)

Here is the result of 'npm run postinstall'. Furthermore I attach 'install.js' and npm-debug.log'
[npm-debug.log.txt](https://github.com/bloomberg/bucklescript/files/714520/npm-debug.log.txt)
[install.js.txt](https://github.com/bloomberg/bucklescript/files/714522/install.js.txt)



../bin/bsc.exe  -no-alias-deps -bs-no-version-header -bs-diagnose -bs-no-check-div-by-zero -bs-cross-module-opt -bs-package-name bs-platform -bs-package-output commonjs:lib/js -bs-package-output goog:lib/goog -bs-package-output amdjs:lib/amdjs -I ../stdlib -nostdlib -nopervasives -open Pervasives -unsafe -w -40-49 -bin-annot  -c caml_weak.ml
../bin/bsc.exe  -no-alias-deps -bs-no-version-header -bs-diagnose -bs-no-check-div-by-zero -bs-cross-module-opt -bs-package-name bs-platform -bs-package-output commonjs:lib/js -bs-package-output goog:lib/goog -bs-package-output amdjs:lib/amdjs -I ../stdlib -nostdlib -nopervasives -open Pervasives -unsafe -w -40-49 -bin-annot  -c caml_module.ml
File "caml_int64.ml", line 251, characters 31-45:
Error: Integer literal exceeds the range of representable integers of type nativeint
make[2]: *** [Makefile:60: caml_int64.cmj] Error 2
make[2]: *** Waiting for unfinished jobs....
Building caml_weak.ml
Building caml_module.ml
make[2]: Leaving directory '/home/helmut/node_modules/bs-platform/jscomp/runtime'
make[1]: *** [Makefile:436: libs] Error 2
make[1]: Leaving directory '/home/helmut/node_modules/bs-platform/jscomp'
make: *** [Makefile:467: world] Error 2
child_process.js:531
    throw err;
    ^

Error: Command failed: make world
    at checkExecSyncError (child_process.js:488:13)
    at Object.execSync (child_process.js:528:13)
    at Object.<anonymous> (/home/helmut/node_modules/bs-platform/scripts/install.js:59:23)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.runMain (module.js:605:10)
    at run (bootstrap_node.js:420:7)

npm ERR! Linux 4.8.11-1-ARCH
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "run" "postinstall"
npm ERR! node v7.2.0
npm ERR! npm  v4.0.3
npm ERR! code ELIFECYCLE
npm ERR! bs-platform@1.4.1 postinstall: `node scripts/install.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the bs-platform@1.4.1 postinstall script 'node scripts/install.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the bs-platform package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node scripts/install.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs bs-platform
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls bs-platform
npm ERR! There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

npm ERR! Please include the following file with any support request:
npm ERR!     /home/helmut/node_modules/bs-platform/npm-debug.log

--- a/jscomp/runtime/caml_int64.ml
+++ b/jscomp/runtime/caml_int64.ml
@@ -259,9 +259,8 @@ let gt x y =
 let le x y = Pervasives.not (gt x y)
 
 
-let to_float ({lo; hi } : t) : float =
-  let x = Nativeint.to_float 0x1_0000n in (* trick to allow compilation on 32-bit systems *)
-    Nativeint.to_float hi *. x *. x +. Nativeint.to_float lo
+let to_float ({hi; lo} : t) = 
+  Nativeint.to_float ( hi *~ [%raw{|0x100000000|}] +~ lo)
 
 
 

--- a/jscomp/runtime/caml_int64.ml
+++ b/jscomp/runtime/caml_int64.ml
@@ -260,7 +260,8 @@ let le x y = Pervasives.not (gt x y)
 
 
 let to_float ({lo; hi } : t) : float =
-  Nativeint.to_float ( hi *~   0x1_0000_0000n +~  lo)
+  let x = Nativeint.to_float 0x1_0000n in (* trick to allow compilation on 32-bit systems *)
+    Nativeint.to_float hi *. x *. x +. Nativeint.to_float lo
 
 
 

